src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/ConfingurationBindingSourceGeneratorTests.cs

```diff
@@ -114,6 +114,45 @@ public class MyClass
             await VerifyAgainstBaselineUsingFile("TestConfigureCallGen.generated.txt", testSourceCode);
         }
 
+        [Fact]
+        public async Task TestBaseline_TestPrimitivesGen()
+        {
+            string testSourceCode = """
+                using System;
+                using System.Collections.Generic;
+                using System.Globalization;
+                using Microsoft.Extensions.Configuration;
+
+                public class Program
+                {
+                    public static void Main()
+                    {
+                        ConfigurationBuilder configurationBuilder = new();
+                        IConfigurationRoot config = configurationBuilder.Build();
+
+                        MyClass options = config.Get<MyClass>();
+                    }
+
+                    public class MyClass
```

Comment:
What about all the other types that TypeConverter supports in the box?

[https://github.com/dotnet/runtime/blob/5711ca33f8dd1758e18c09ebdedf7e95eb424e0f/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs#L164-L191](https://github.com/dotnet/runtime/blob/5711ca33f8dd1758e18c09ebdedf7e95eb424e0f/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs#L164-L191)

Guid, Version, Date/Time types, float, double, etc.

Reply:
I am seeing these types are supported in the parser. I agree will be good to add them to the test here too. good idea!

Reply:
There are unit tests for all of the types in question.

For the baseline tests, I cherry-picked representative types that cover all the parsing schemes used in the emitter. But yeah I'll add all of them to catch regressions.